application/.github/workflows/test-and-publish.yml

```
@@ -0,0 +1,59 @@
+name: Test
+
+on:
+  push:
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Run unit tests
+        run: |
+          docker-compose -f actions-services.yml run --rm app ./run-tests.sh
```

Technically, these one liners don't need the `|` and new line, but are fine either way. I might also put a new line between each job.
```suggestion
        run: docker-compose -f actions-services.yml run --rm app ./run-tests.sh

```